src/components/ListItem.jsx

```diff
@@ -57,22 +97,100 @@ export function ListItem({ name, listToken, itemId, data, urgency }) {
 		}
 	};
 
+	const computeDate = (secs) => {
+		const output = new Date(secs * 1000).toString().split(' ');
+		return `${output[1]} ${output[2]}, ${output[3]}`;
+	};
+
+	const lastPurchase = data.dateLastPurchased
+		? computeDate(data.dateLastPurchased.seconds)
+		: 'N/A';
+	const nextPurchase = data.dateNextPurchased
+		? computeDate(data.dateNextPurchased.seconds)
```

Comment:
Same goes for this line of code, can use optional chaining to return undefined instead of 'N/A'
```suggestion
		? computeDate(data.dateNextPurchased?.seconds)
```